component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidator.java

```diff
@@ -283,6 +286,35 @@ private boolean validateJWTInDataBase(String jti, long currentTimeInMillis,
         return true;
     }
 
+    /**
+     * This method is to get JTI from the DB.
+     * For the migration purposes (preserve existing behaviour),
+     * We are searching for the current tenant and default tenant.
+     *
+     * @param jti      JTI.
+     * @param tenantId Tenant id.
+     * @return JWT entry if exists.
+     * @throws OAuthClientAuthnException OAuthClientAuthnException.
+     */
+    private JWTEntry getJTIfromDB(String jti, int tenantId) throws OAuthClientAuthnException {
+
+        List<JWTEntry> jwtEntries = jwtStorageManager.getJwtsFromDB(jti, tenantId);
+        if (jwtEntries.isEmpty()) {
+            return null;
+        }
+        // If there is only one entry return it.
+        else if (jwtEntries.size() == 1) {
```

Comment:
Having multiple if, else blocks here seems to be not required. Shall we add a for each loop ?

Reply:
If loop simplified